src/Components/SmartComponents/Reports/Common/firstPagePDF.js

```diff
@@ -4,55 +4,58 @@ import { Paragraph } from '@redhat-cloud-services/frontend-components-pdf-genera
-                date: filtersBuilder.publish_date.values,
-                b: (...chunks) => <Text style={{ fontWeight: 700 }}>{chunks}</Text>
-            })}
+        <Paragraph>
+            {filtersBuilder.hasOwnProperty('filter')
+                ? intl.formatMessage(messages.customReportSearchTermCvssAndPublishDate, {
+                    search_term: filtersBuilder.filter.values,
+                    score: (filtersBuilder.cvss_filter && filtersBuilder.cvss_filter.values.toLowerCase()) || '0.0 - 10.0',
+                    published: (filtersBuilder.publish_date && filtersBuilder.publish_date.values)
+                        || intl.formatMessage(messages.customReportTimespanAnytime,
+                            { b: (...chunks) => <Text style={{ fontWeight: 700 }}>{chunks}</Text> }),
+                    b: (...chunks) => <Text style={{ fontWeight: 700 }}>{chunks}</Text>
+                })
+                : intl.formatMessage(messages.customReportCvssAndPublishDate, {
+                    score: (filtersBuilder.cvss_filter && filtersBuilder.cvss_filter.values.toLowerCase()) || '0.0 - 10.0',
+                    published: (filtersBuilder.publish_date && filtersBuilder.publish_date.values)
+                        || intl.formatMessage(messages.customReportTimespanAnytime,
+                            { b: (...chunks) => <Text style={{ fontWeight: 700 }}>{chunks}</Text> }),
+                    b: (...chunks) => <Text style={{ fontWeight: 700 }}>{chunks}</Text>
+                })
+            }
             {filtersBuilder.hasOwnProperty('impact') && intl.formatMessage(messages.CVEsPDFreportFiltersSeverity, {
-                severity: filtersBuilder.impact.values,
+                severity: filtersBuilder.impact.values.toLowerCase(),
                 b: (...chunks) => <Text style={{ fontWeight: 700 }}>{chunks}</Text>
             })}
             {filtersBuilder.hasOwnProperty('business_risk_id') && intl.formatMessage(messages.CVEsPDFreportFiltersBusinessRisk, {
-                businessRisk: filtersBuilder.business_risk_id.values,
+                businessRisk: filtersBuilder.business_risk_id.values.toLowerCase(),
                 b: (...chunks) => <Text style={{ fontWeight: 700 }}>{chunks}</Text>
             })}
             {filtersBuilder.hasOwnProperty('status_id') && intl.formatMessage(messages.CVEsPDFreportFiltersStatus, {
-                status: filtersBuilder.status_id.values,
+                status: filtersBuilder.status_id.values.toLowerCase(),
                 b: (...chunks) => <Text style={{ fontWeight: 700 }}>{chunks}</Text>
             })}
             {filtersBuilder.hasOwnProperty('security_rule') && intl.formatMessage(messages.CVEsPDFreportFiltersSecurityRule, {
-                securityRule: filtersBuilder.security_rule.values,
+                securityRule: filtersBuilder.security_rule.values.toLowerCase(),
```

Comment:
that's a great example when we need to start thinking about the next phase (after making it work, what's next? ) 

You have a block of code that has the filters and by adding a simple chunk of code (`toLowerCase`) you had to modify multiple lines. In that case, there are only few filters but what if we have several in the future? Some other observations

-  repetitive code. It follows a pattern like  `checkStatement && TranslationID, {key: value}, formating` a lot of times
-  adding `;` in the translation can be easily delete
-  you're not able to determine which filter is the last one (so you would be able to add the `and` in the end)

Reply:
you can easily fix the problem above by generating the filters dynamically(code). For example, have a mapping and loop over it  to determine if filter exists and display it...and so on..